lib/runtime/index.js

```diff
@@ -12,9 +12,12 @@ var Entry = require("./entry.js");
 // specific module objects, or for Module.prototype (where implemented),
 // to make the runtime available throughout the entire module system.
 exports.enable = function (mod) {
+  if (typeof mod.resolve !== "function") {
+    throw new Error("module.resolve not implemented");
+  }
```

Comment:
Not all module systems implement a `Module` constructor with a shared `Module.prototype`, and only some of those module systems implement `Module.prototype.resolve`. Since `module.link` now calls `module.resolve(id)` dynamically, a working implementation of `module.resolve` is now required. This is a significant requirement for sure, but hardly the deal-breaker that I once imagined it to be.